components/canvas/raqote_backend.rs

```diff
@@ -225,16 +211,18 @@ impl GenericDrawTarget for raqote::DrawTarget {
     }
     fn create_source_surface_from_data(&self, data: Snapshot) -> Option<Self::SourceSurface> {
         Some(
-            data.to_vec(
-                Some(SnapshotAlphaMode::Transparent {
-                    premultiplied: true,
-                }),
-                Some(SnapshotPixelFormat::BGRA),
+            bytemuck::try_cast_vec(
+                data.to_vec(
+                    Some(SnapshotAlphaMode::Transparent {
+                        premultiplied: true,
+                    }),
+                    Some(SnapshotPixelFormat::BGRA),
+                )
+                .0,
             )
-            .0,
+            .unwrap_or_else(|(_, surface)| bytemuck::pod_collect_to_vec(&surface)),
```

Comment:
we fallback to new alloc if vec is not properly aligned. before we always did new alloc.